Base/usr/share/man/man1/test.md

```diff
@@ -0,0 +1,86 @@
+## Name
+
+test - checks files and compare values
+
+## Synopsis
+
+```**sh
+$ test EXPRESSION
+```
+
+## Description
+
+`test` takes a given expression and returns 0 if the expression evaluates to true, or 1 if not.
+
+An omitted expression defaults to false, and an unexpected error causes a return value of 126.
+
+The expression can take any of the following forms:
+
+* `( EXPRESSION )` (grouping)
+
+Boolean operations
+
+* `! EXPRESSION` (negation)
+* `EXPRESSION -a EXPRESSION` (and)
+* `EXPRESSION -o EXPRESSION` (or)
+
+String comparison
+
+* `-n STRING` (non-empty)
+* `-z STRING` (empty)
+* `STRING = STRING` (equality)
+* `STRING != STRING` (unequality)
+* `STRING` (non-empty, if the only expression available)
+
+Integer comparison
+
+* `INTEGER -eq INTEGER` (equality)
+* `INTEGER -ne INTEGER` (unequality)
+* `INTEGER -lt INTEGER` (less-than)
+* `INTEGER -gt INTEGER` (greater-than)
+* `INTEGER -le INTEGER` (less-than-or-equal)
+* `INTEGER -ge INTEGER` (greater-than-or-equal)
+
+File comparison
+
+* `FILE -ef FILE` (same file - same device and inode number)
+* `FILE -nt FILE` (newer-than - checks modification date)
+* `FILE -ot FILE` (older-than - checks modification date)
+
+File type checks
+
+* `-b FILE` (is block device?)
+* `-c FILE` (is character device?)
+* `-f FILE` (is regular file?)
+* `-d FILE` (is directory?)
+* `-p FILE` (is pipe?)
+* `-S FILE` (is socket?)
+* `-h FILE` (is symbolic link?)
+* `-L FILE` (is symbolic link?)
+
+File permission checks
+
+* `-r FILE` (is readable?)
+* `-w FILE` (is writable?)
+* `-x FILE` (is executable?)
```

Comment:
I feel like this is not communicated clear enough, neither in the man page nor in the code: this is not a check whether the file is readable/writable/executable, it's whether the *current user* has permissions to read/write/execute it.

Reply:
I'm not sure how much more clear the code can get, it's literally called `HasPermission::Read`

Reply:
"Has permission" still sounds like we're checking if the file has this permission (or at least it doesn't sound sufficiently unlike that). `UserHasPermission` or something.